config/config-package.py

```diff
@@ -550,6 +550,38 @@ def manifest_in(self):
                 manifest_additional_rules=manifest_additional_rules,
                 with_docs=self.with_docs)
 
+    def pyproject_toml(self):
+        """Modify pyproject.toml with meta options."""
+        pyproject_toml_path = self.path / 'pyproject.toml'
+        if pyproject_toml_path.exists():
+            with open(pyproject_toml_path, 'rb') as fp:
+                pyproject_data = tomlkit.load(fp)
+            pyproject_toml = collections.defaultdict(dict, **pyproject_data)
+        else:
+            pyproject_toml = collections.defaultdict(dict)
+
+        buildsystem_cfg = pyproject_toml['build-system']
+
+        # Insert sane default for build-system:requires
+        requires = buildsystem_cfg.get('requires', [])
+        setuptools_requirement = [
+            x for x in requires if x.startswith('setuptools')]
+        if not setuptools_requirement:
+            requires.append(f'setuptools{SETUPTOOLS_VERSION_SPEC}')
+            buildsystem_cfg['requires'] = requires
```

Comment:
How does it handle updating `SETUPTOOLS_VERSION_SPEC` to a newer version number?
If I understood the code correctly, it only writes the version spec once as long it does not exist.
Did I miss something here?

Reply:
You're correct, I did not take that into account. All I thought was "don't overwrite existing setuptools specs, the author will have had a good reason to put it there". I changed it to remove all setuptools specs it finds and just add our own.

Successfully tested against Zope and BTrees. Please re-check.